flujo/application/core/step_policies.py

```
@@ -4721,6 +4829,15 @@ def _collect_names_recursive(p: Any) -> None:
 
                 try:
                     core._enable_cache = False
+                    
+                    # CRITICAL FIX FOR RESUME: When resuming, restore the last output as current_data
+                    # The 'data' parameter contains human input on resume, which is wrong for the loop
+                    if is_resuming and saved_last_output is not None:
+                        current_data = saved_last_output
+                        telemetry.logfire.info(
+                            f"[POLICY] RESUME: Restored loop data from saved state, human input will be passed to HITL step"
```

Ruff (0.13.3)

f-string without any placeholders

Remove extraneous `f` prefix

(F541)